source/reference/minio-mc/mc-ilm-add.rst

```diff
@@ -238,26 +235,55 @@ Global Flags
 Examples
 --------
 
-Expire Bucket Contents After Number of Days
-~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
+Expire All Bucket Contents After Number of Days
+~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-Use :mc:`mc ilm add` with :mc-cmd:`~mc ilm add --expiry-days` to
-expire bucket contents a number of days after object creation:
+Use :mc:`mc ilm add` with :mc-cmd:`~mc ilm add --expiry-days` to expire bucket contents a number of days after object creation:
 
 .. code-block:: shell
    :class: copyable
 
    mc ilm add ALIAS/PATH --expiry-days "DAYS" 
```

Comment:
Alas, not quite. This will create delete markers, but not actually delete the objects. You have to include both --noncurrent-expiration-days and  --expired-object-delete-marker flag to also delete versioned objects + tombstones.

This does work for non-versioned buckets though. Will leave it to you if you want to try addressing both.

Reply:
Hmmm. I tweaked the explanation some to emphasize that the objects are marked for expiration.